satellite/notification/service.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notification
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/private/sync2"
+	"storj.io/storj/satellite/mailservice"
+	"storj.io/storj/satellite/overlay"
+)
+
+type ClientSetting struct {
+	Emails    int
+	RPC       int
+	lastReset time.Time
+}
+
+// Service is the notification service between storage nodes and satellites.
+// architecture: Service
+type Service struct {
+	log     *zap.Logger
+	config  Config
+	dialer  rpc.Dialer
+	overlay *overlay.Service
+	mailer  *mailservice.Service
+
+	loop    *sync2.Cycle
+	lock    *sync.Mutex
+	limiter map[string]ClientSetting
```

Comment:
Lets get rid of the limiter for now, as we intend to implement a proper rate limiting for other packages as well.
We will just send as many requests for now as they come up.